rubix-core/src/main/java/com/qubole/rubix/core/CachingFileSystem.java

```diff
@@ -206,7 +219,27 @@ public boolean mkdirs(Path path, FsPermission fsPermission)
     public FileStatus getFileStatus(Path path)
             throws IOException
     {
-        return fs.getFileStatus(path);
+        FileStatus originalStatus = fs.getFileStatus(path);
```

Comment:
So changing listing's path to rubix scheme was the main change to support rubix scheme!!

Reply:
Yes this one is main along with initializing  the remotefilesystem with appropriate scheme (for CachingNativeS3FileSystem, its s3n) so that when we are reading from the remote filesystem, it gets reflected properly